requirements/dev.in

```diff
@@ -1,2 +1,3 @@
--r base.txt
```

Comment:
Normally, you need to include `base.in` to properly layer the direct requirements. You can actually add `-c base.txt` inside *that file*, and have an additional `-c dev.txt` inside this one. Adding constraints from other envs might cause confusion, I think.

Reply:
I view `base` requirements as core necessities, with `dev` (and potentially  `linters`, `prod` or others later on) addressing according "additional" needs. This approach enables me to set up dependencies like building blocks, depending on use case objectives.

Reply:
This layering seems fine. Except that sometimes these things don't have to be related (like some linters may not really need all of your app deps to be installed and could be entirely separate, depending on what is set up and how).

Reply:
It's just that you shouldn't generally entangle pip-tools' input files (with the direct dependency lists) and the output files (constraints/lockfiles). Don't make *inputs* of one env depend on the *outputs* of some other.